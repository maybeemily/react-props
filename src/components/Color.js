import React from 'react';
import PropTypes from 'prop-types';

function Color({ name, hex, rgb }) {
  const style = {
    'background-color' : hex 
  };

  return (
    <ul>
      <li>Color Name: <span style={style}>{name ? name : hex}</span></li>
      <li>Color Hex: {hex}</li>
      <li>Color RGB: {rgb}</li>
    </ul>
  );
}

Color.propTypes = {
  name: PropTypes.string,
  hex: PropTypes.string.isRequired,
  rgb: PropTypes.string.isRequired
};

export default Color;
