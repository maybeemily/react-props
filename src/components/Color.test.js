import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color';

describe('Color test', () => {
  it('renders the color', () => {
    const wrapper = shallow(<Color hex="#c6f470" rgb="(198, 244, 112)" />);
    expect(wrapper).toMatchSnapshot();
  });
});
