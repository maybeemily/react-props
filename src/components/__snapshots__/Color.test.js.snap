// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color test renders the color 1`] = `
<ul>
  <li>
    Color Name: 
    <span
      style={
        Object {
          "background-color": Object {
            "hex": "#c6f470",
          },
        }
      }
    >
      #c6f470
    </span>
  </li>
  <li>
    Color Hex: 
    #c6f470
  </li>
  <li>
    Color RGB: 
    (198, 244, 112)
  </li>
</ul>
`;
